### ISSUE TITLE
Opening app offline causes crash

### ISSUE THREAD

**Issue by erlend-sh** [spicylobstergames/SpicyLauncher#23]
I just tried to start up the launcher while my wifi was off. Instead of some kind of "can't connect to internet" notification, the app just crashes.

```
System Integrity Protection: enabled

Crashed Thread:        18  tokio-runtime-worker
Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Application Specific Information:
abort() called
```

Easiest solution here is the aforementioned no-connection notice. In the future we could consider making the app work offline, i.e. it'll still give you access to whatever you've already downloaded even if you're offline.

**Comment by zicklag** [comment]
> In the future we could consider making the app work offline, i.e. it'll still give you access to whatever you've already downloaded even if you're offline.

Yeah, huge 👍 for that. It drives me crazy when apps don't let me get to what I have on my own omputer when I'm offline.

Maybe we could cache the release files locally whenever we download them, and then we show like an "out-of-date" or "disconnected" warning icon with a tooltip somewhere a little bit out of the way or something.

**Comment by orhun** [comment]
Agree with @zicklag here. Caching + showing a notification would be nice.
